File: OverseeGateway/TestPostAndGet/Form1.vb
Public Class frmPosAndGet
    Private Sub btnPost_Click(sender As Object, e As EventArgs) Handles btnPost.Click
        Dim Name As String = "POST"
        Dim password As String = "Your Passord"
        'Dim postString As String = String.Format("Name={0}&Password={1}", Name, password)
        Dim postString As String = String.Format("Name={0}", Name)
        Dim req As System.Net.HttpWebRequest = System.Net.HttpWebRequest.Create("http://localhost/default.aspx")
        Dim contentType As String = "application/x-www-form-urlencoded"
        req.Method = System.Net.WebRequestMethods.Http.Post
        req.ContentType = contentType
        req.ContentLength = System.Text.Encoding.UTF8.GetByteCount(postString)
        req.Timeout = 30000
        req.Accept = "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"
        Dim requestWriter As System.IO.StreamWriter = New System.IO.StreamWriter(req.GetRequestStream)

        Try
            requestWriter.Write(postString)
            requestWriter.Flush()
            requestWriter.Close()
        Catch ex As Exception
            MessageBox.Show(ex.ToString)
        End Try
        Dim response As System.Net.WebResponse = req.GetResponse



        If CType(response, System.Net.HttpWebResponse).StatusCode = Net.HttpStatusCode.OK Then
            Dim responseReader As System.IO.StreamReader = New System.IO.StreamReader(response.GetResponseStream)

            Dim responseData As String = responseReader.ReadToEnd

            responseReader.Close()
            req.GetResponse.Close()
        Else

        End If
    End Sub
    Private Sub btnGet_Click(sender As Object, e As EventArgs) Handles btnGet.Click
        Dim Name As String = "GET1"
        Dim password As String = "Your Passord"
        'Dim postString As String = String.Format("Name={0}&Password={1}", Name, password)
        Dim postString As String = String.Format("Name={0}", Name)
        Dim url As String = "http://localhost/default.aspx"
        If Not String.IsNullOrEmpty(postString) Then
            url = url & "?" & postString
        End If
        Dim req As System.Net.HttpWebRequest = System.Net.HttpWebRequest.Create(url)

        Dim contentType As String = "application/x-www-form-urlencoded"
        req.Method = System.Net.WebRequestMethods.Http.Get

        req.ContentType = contentType

        req.Timeout = req.Timeout
        'req.Accept = "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"

        Dim response As System.Net.WebResponse = req.GetResponse



        If CType(response, System.Net.HttpWebResponse).StatusCode = Net.HttpStatusCode.OK Then
            Dim responseReader As System.IO.StreamReader = New System.IO.StreamReader(response.GetResponseStream)

            Dim responseData As String = responseReader.ReadToEnd

            responseReader.Close()
            req.GetResponse.Close()
        Else

        End If
    End Sub
End Class

File: CableSoft Developer/IVRCaller/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.IO;
using System.Net;
using System.Windows.Forms;

namespace IVRCaller
{
    public partial class fmMain : Form
    {
        public fmMain()
        {
            InitializeComponent();
        }

        private void btnSend_Click(object sender, EventArgs e)
        {
            string strXMLFile = @"D:\CableSoft Document\IVR_Net\XML Sample.xml";
            string strTxt = File.ReadAllText(strXMLFile);
            strTxt = txtUrl.Text + "?ID=" + strTxt;
            System.Net.HttpWebRequest aRequest = (System.Net.HttpWebRequest)System.Net.HttpWebRequest.Create( strTxt   );
            aRequest.Method = "POST";
            aRequest.Timeout = 1000;
            aRequest.Credentials = System.Net.CredentialCache.DefaultCredentials;
            System.Net.HttpWebResponse aResponse = (System.Net.HttpWebResponse)aRequest.GetResponse();
            System.IO.Stream aStream = aResponse.GetResponseStream();
            System.IO.StreamReader aReader = new System.IO.StreamReader( aStream, Encoding.UTF8 );
            PostRecver.Text = aReader.ReadToEnd();
            aStream.Close();
            aReader.Close();
            aResponse.Close();
            MessageBox.Show("完成");
        }


        private void btnSend2_Click(object sender, EventArgs e)
        {
            //string strXMLFile = @"D:\CableSoft Document\IVR_Net\XML Sample.xml";
            //string strTxt = "?Company=3&Language=1&Func=2&History=2,1&Tel=8525718&Floor=3&CMD=N21";
            //string strTxt = "?Company=5&Language=1&Func=2&History=2,1&Tel=8357828&Floor=3&CMD=N21";
            HttpWebResponse aTest;
            try
            {
                string strTxt = string.Empty;
                
                strTxt = txtUrl.Text + txtPara.Text;
                //strTxt = textBox1.Text;
                System.Net.HttpWebRequest aRequest = (System.Net.HttpWebRequest)System.Net.HttpWebRequest.Create(strTxt);
                aRequest.Method = "POST";
                aRequest.Timeout = 1000;
                aRequest.Credentials = System.Net.CredentialCache.DefaultCredentials;
                System.Net.HttpWebResponse aResponse = (System.Net.HttpWebResponse)aRequest.GetResponse();
                aTest = aResponse;
                System.IO.Stream aStream = aResponse.GetResponseStream();
                System.IO.StreamReader aReader = new System.IO.StreamReader(aStream, Encoding.UTF8);
                PostRecver.Text = aReader.ReadToEnd();
                aStream.Close();
                aReader.Close();
                aResponse.Close();

                MessageBox.Show("完成");
            }
            catch (WebException  ex)
            {
                if (ex.Status == WebExceptionStatus.Timeout)
                {

                    MessageBox.Show("TimeOut");

                }
                else
                {
                    MessageBox.Show(ex.Message.ToString());
                    
                }

                
                    
                
                
            }


            
        }

        private void btnSend3_Click(object sender, EventArgs e)
        {
            string strTxt = string.Empty;
            strTxt = txtUrl.Text + txtPara.Text;
            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(strTxt);
            request.MaximumAutomaticRedirections = 4;
            request.MaximumResponseHeadersLength = 4;
            // Set credentials to use for this request.
            request.Credentials = CredentialCache.DefaultCredentials;
            HttpWebResponse response = (HttpWebResponse)request.GetResponse();
            PostRecver.Text = response.Headers.ToString();
            MessageBox.Show("完成");

            /*
                Console.WriteLine("Content length is {0}", response.ContentLength);
                Console.WriteLine("Content type is {0}", response.ContentType);

                // Get the stream associated with the response.
                Stream receiveStream = response.GetResponseStream();

                // Pipes the stream to a higher level stream reader with the required encoding format. 
                StreamReader readStream = new StreamReader(receiveStream, Encoding.UTF8);

                Console.WriteLine("Response stream received.");
                Console.WriteLine(readStream.ReadToEnd());
                response.Close();
                readStream.Close();
            */

        }

        

    }
}